libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity/Scripts/Media/AudioReceiver.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.WebRTC.Unity
+{
+    /// <summary>
+    /// This component represents a remote audio source added as an audio track to an
+    /// existing WebRTC peer connection by a remote peer and received locally.
+    /// The audio track can optionally be displayed locally with a <see cref="MediaPlayer"/>.
+    /// </summary>
+    [AddComponentMenu("MixedReality-WebRTC/Audio Receiver")]
+    public class AudioReceiver : MediaReceiver, IAudioSource
+    {
+        public bool IsStreaming { get; protected set; }
+
+        public AudioStreamStartedEvent AudioStreamStarted = new AudioStreamStartedEvent();
+        public AudioStreamStoppedEvent AudioStreamStopped = new AudioStreamStoppedEvent();
+
+        public AudioStreamStartedEvent GetAudioStreamStarted() { return AudioStreamStarted; }
+        public AudioStreamStoppedEvent GetAudioStreamStopped() { return AudioStreamStopped; }
+
+        /// <summary>
+        /// Audio transceiver this receiver is paired with.
+        /// This is <c>null</c> until a remote description is applied which pairs the receiver
+        /// with the remote track of the transceiver, or until the peer connection associated
+        /// with this receiver creates the audio receiver right before creating an SDP offer.
+        /// </summary>
+        public Transceiver Transceiver { get; private set; }
+
+        /// <summary>
+        /// Remote audio track receiving data from the remote peer.
+        /// This is <c>null</c> until a remote description is applied which pairs the receiver
+        /// with the remote track of the transceiver.
+        /// </summary>
+        public RemoteAudioTrack Track { get; private set; }
+
+        public AudioReceiver() : base(MediaKind.Audio)
+        {
+        }
+
+        /// <summary>
+        /// Register a frame callback to listen to incoming audio data receiving through this
+        /// audio receiver from the remote peer.
+        /// The callback can only be registered once the <see cref="Track"/> is valid, that is
+        /// once the <see cref="AudioStreamStarted"/> event was triggered.
+        /// </summary>
+        /// <param name="callback">The new frame callback to register.</param>
+        /// <remarks>
+        /// A typical application might use this callback to display some feedback of a local webcam recording.
+        /// 
+        /// Note that registering a callback does not influence the video capture and sending to the
+        /// remote peer, which occurs whether or not a callback is registered.
+        /// </remarks>
+        public void RegisterCallback(AudioFrameDelegate callback) { }
```

Comment:
Add a TODO saying that this is not doing anything?

Reply:
Actually this was missing. The one on `AudioSender` does nothing though, so is now throwing a `NotImplementedException`. Also added a bunch of comments and warnings showing up prominently in the docs about all the audio limitations until we fix them.